valhalla/sif/dynamiccost.h

```diff
@@ -186,6 +197,32 @@ class DynamicCost {
     return (node->access() & access_mask_) || ignore_access_;
   }
 
+  /**
+   * Used for determine the viability of a candidate edge as well as a conservative reachability
+   * The notable difference to the full featured allowed method is this methods lack of info
+   * about the currently tracked path (hence why its conservative)
+   *
+   * This method is to be used by loki::search and loki::reach and the base class implementation is
+   * used to do basically accessibility and adherence to the disallow mask
+   *
+   * @param edge           the edge that should or shouldnt be allowed
+   * @param tile           the tile which contains the edge (for traffic lookup)
+   * @param disallow_mask  a mask that controls additional properties that should disallow the edge
+   * @return true if the edge is allowed to be used (either as a candidate or a reach traversal)
+   */
+  inline virtual bool Allowed(const baldr::DirectedEdge* edge,
```

Comment:
i renamed the `Filter` function to be `Allowed` like the other functions for nodes and pairs of edges on a path. I've also provided a default imlementation that does some basic things like checking access and doing the conservative rejection that that reach method needs. This allowed me to greatly simplify downstream costings' implementations, one day we wont even need to implement them but thats a story for another time.